From f534c9a28e2adeeb8e49c32eeb4ba56c7d07aaa2 Mon Sep 17 00:00:00 2001
From: Liu Jian <liujian56@huawei.com>
Date: Tue, 10 May 2022 23:05:05 +0800
Subject: [PATCH] bpf: Access bpf_sock's src_ip4 and sorc_port in
 BPF_CGROUP_INET_SOCK_RELEASE hook
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f534c9a28e2adeeb8e49c32eeb4ba56c7d07aaa2
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I545NW
CVE: NA

--------------------------------

Access bpf_sock's src_ip4 and sorc_port in BPF_CGROUP_INET_SOCK_RELEASE
hook.

Signed-off-by: Liu Jian <liujian56@huawei.com>
Reviewed-by: Wei Yongjun <weiyongjun1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 net/core/filter.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/core/filter.c b/net/core/filter.c
index 61cb3f94bd03..fa473a58d1be 100644
--- a/net/core/filter.c
+++ b/net/core/filter.c
@@ -7742,6 +7742,7 @@ static bool __sock_filter_check_attach_type(int off,
 	case bpf_ctx_range(struct bpf_sock, src_ip4):
 		switch (attach_type) {
 		case BPF_CGROUP_INET4_POST_BIND:
+		case BPF_CGROUP_INET_SOCK_RELEASE:
 			goto read_only;
 		default:
 			return false;
@@ -7757,6 +7758,7 @@ static bool __sock_filter_check_attach_type(int off,
 		switch (attach_type) {
 		case BPF_CGROUP_INET4_POST_BIND:
 		case BPF_CGROUP_INET6_POST_BIND:
+		case BPF_CGROUP_INET_SOCK_RELEASE:
 			goto read_only;
 		default:
 			return false;
-- 
2.33.0

